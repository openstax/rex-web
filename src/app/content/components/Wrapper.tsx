import React from 'react';
import { connect } from 'react-redux';
import styled, { css } from 'styled-components/macro';
import ScrollLock from '../../components/ScrollLock';
import theme from '../../theme';
import { AppState } from '../../types';
import * as selectSearch from '../search/selectors';
import * as contentSelectors from '../selectors';
import { contentWrapperMaxWidth, verticalNavbarMaxWidth } from './constants';

export { wrapperPadding } from '../../components/Layout';

interface WrapperProps {
  hasQuery: boolean;
  verticalNavOpen: boolean;
  className?: string;
}

// tslint:disable-next-line:variable-name
export const Wrapper = styled(
  ({hasQuery, verticalNavOpen, children, ...props}: React.PropsWithChildren<WrapperProps>) =>
    <ContentLayoutBody {...props}>
      {verticalNavOpen && <ScrollLock overlay={false} mediumScreensOnly={true} />}
      {children}
    </ContentLayoutBody>
)`
  position: relative; /* for sidebar overlay */
  overflow: visible; /* so sidebar position: sticky works */

  @media screen {
    ${theme.breakpoints.mobile(css`
      margin-left: 0;
    `)}
  }
`;

// tslint:disable-next-line:variable-name
const ContentLayoutBody = styled.div`
  width: 100%;
  max-width: ${contentWrapperMaxWidth + verticalNavbarMaxWidth * 2}rem;
  margin: 0 auto;
  display: grid;
  grid-template-columns: 8rem auto auto;
  ${theme.breakpoints.mobileMedium(css`
    grid-template-columns: 100%;
  `)}
`;

export default connect(
  (state: AppState) => ({
    hasQuery: !!selectSearch.query(state),
    verticalNavOpen: contentSelectors.mobileMenuOpen(state) || selectSearch.searchResultsOpen(state),
  })
)(Wrapper);
