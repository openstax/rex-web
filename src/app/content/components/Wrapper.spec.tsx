import React from 'react';
import renderer from 'react-test-renderer';
import { Wrapper } from './Wrapper';

describe('wrapper', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(<Wrapper />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot with search open', () => {
    const tree = renderer.create(<Wrapper hasQuery verticalNavOpen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
