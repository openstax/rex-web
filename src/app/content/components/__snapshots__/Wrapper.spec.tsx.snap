// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wrapper matches snapshot 1`] = `
<div
  className="Wrapper-m6kdbg-0 bwERCu Wrapper__ContentLayoutBody-m6kdbg-1 fRDSoc"
/>
`;

exports[`wrapper matches snapshot with search open 1`] = `
<div
  className="Wrapper-m6kdbg-0 bwERCu Wrapper__ContentLayoutBody-m6kdbg-1 fRDSoc"
/>
`;
